### PR TITLE
feat: add BitVec.toNat_[udiv|umod] and [udiv|umod]_eq

### PR DESCRIPTION
**Pull request by tobiasgrosser**

### CODE REVIEW

File: src/Init/Data/BitVec/Lemmas.lean [hunk 1]
```diff
@@ -857,6 +857,31 @@ theorem sshiftRight_add {x : BitVec w} {m n : Nat} :
...
+/-! ### umod -/
+
+theorem umod_eq {x y : BitVec n} :
+    x.umod y = BitVec.ofNat n (x.toNat % y.toNat) := by
+  have h : x.toNat % y.toNat < 2 ^ n := Nat.lt_of_le_of_lt (Nat.mod_le _ _) x.isLt
+  simp [umod, bv_toNat, Nat.mod_eq_of_lt h]
+
+@[simp, bv_toNat]
+theorem toNat_umod {x y : BitVec n} :
+    (x.umod y).toNat = x.toNat % y.toNat := by rfl
```
Comment (AtticusKuhn) [inline src/Init/Data/BitVec/Lemmas.lean:883]:
```suggestion
    (x.umod y).toNat = x.toNat % y.toNat := rfl
```

It's up to you, but it seems that "by" is not necessary with "rfl"

https://github.com/leanprover/lean4/blob/5c182bd5402653a6bd8b89fb7b0ece3d4670b5e7/src/Init/Data/List/Impl.lean#L188

**Review by AtticusKuhn — APPROVED** [review]
Looks good to me.

**Review by AnotherAlexHere — APPROVED** [review]
LGTM.

**Comment by tobiasgrosser** [comment]
I am surprised this breaks mathlib. Anyone knows why?

**Comment by semorrison** [comment]
Sorry, Mathlib breakage is spurious, CI has been complicated recently...

**Pull request merged**